assignments/mergeIntervals/lc252/lc252.js

```diff
@@ -13,9 +13,31 @@ class Interval {
   }
 }
 
+// Precondition:
+// Intervals: an array of Intervals that represent appointment Times
+// Post condition:
+// canAttendAll: boolean -> The intervals array has no intersecting intervals and the person can attend all appointments
+
+// Sort the interval array by their start value
+// Iterate until the second to last element of the array
+// Make a comparison of the i and i+1 interval
+// Comparison: if intervals[i].end > intervals[i+1].start  RETURN false -> we found an intersecting interval
+// After loop-> we mad it to last interval, then we return true be default because we haven't found any intersecting interval
+
 const can_attend_all_appointments = function(intervals) {
-  // TODO: Write your code here
-  return false;
+
+  const arrLength = intervals.length;
+  if(arrLength === 1) return true;
+  intervals.sort((a, b) => a.start - b.start); // Sort all the intervals by their start value
+  const lastInterval = arrLength - 1; // Last interval that i will make it to
+  let currentIndex = 0; // our i index
+
+  while(currentIndex < lastInterval){
+    if (intervals[currentIndex].end > intervals[currentIndex+1].start) return false; // Checks if there are intersecting invervals
+    currentIndex++;
+  }
+  // No intersecting intervals if the while-loop terminates
+  return true;
 };
```

Comment:
Good work, code is clean! 

Small feedback: if comments help you in code base while working then it is good idea to continue using them, BUT, make sure to delete all non-necessary comments from the code base as it reduces readability.

Where is the love for typing it twice?!?!? #LETSGO #PracticeHard #PlayHard #KeepGrinding